packages/web-components/src/components/ic-tab-context/ic-tab-context.spec.ts

```diff
@@ -0,0 +1,369 @@
+    const page = await newSpecPage({
+      components: [TabContext, TabGroup, Tab, TabPanel],
+      html: `<ic-tab-context>
+      <ic-tab-group label="Example tab group">
+        <ic-tab>One</ic-tab>
+        <ic-tab>Two</ic-tab>
+        <ic-tab>Three</ic-tab>
+      </ic-tab-group>
+      <ic-tab-panel>Tab One</ic-tab-panel>
+      <ic-tab-panel>Tab Two</ic-tab-panel>
+      <ic-tab-panel>Tab Three</ic-tab-panel>
+      </ic-tab-context>`,
+    });
+
+    expect(page.root).toMatchSnapshot();
+
+    //test disconnected callback for automatic activation
+    page.setContent("");
+  });
+
+  it("should render tab context with tabs - light appearance", async () => {
+    const page = await newSpecPage({
+      components: [TabContext, TabGroup, Tab, TabPanel],
+      html: `<ic-tab-context appearance="light">
+      <ic-tab-group label="Example tab group" appearance="light">
+        <ic-tab>One</ic-tab>
+        <ic-tab>Two</ic-tab>
+        <ic-tab>Three</ic-tab>
+      </ic-tab-group>
+      <ic-tab-panel>Tab One</ic-tab-panel>
+      <ic-tab-panel>Tab Two</ic-tab-panel>
+      <ic-tab-panel>Tab Three</ic-tab-panel>
+      </ic-tab-context>`,
+    });
+
+    expect(page.root).toMatchSnapshot();
+  });
+
+  it("should render tab context with tabs - selected tab", async () => {
+    const page = await newSpecPage({
+      components: [TabContext, TabGroup, Tab, TabPanel],
+      html: `<ic-tab-context selected-tab-index="2">
+      <ic-tab-group label="Example tab group">
+        <ic-tab>One</ic-tab>
+        <ic-tab>Two</ic-tab>
+        <ic-tab>Three</ic-tab>
+      </ic-tab-group>
+      <ic-tab-panel>Tab One</ic-tab-panel>
+      <ic-tab-panel>Tab Two</ic-tab-panel>
+      <ic-tab-panel>Tab Three</ic-tab-panel>
+      </ic-tab-context>`,
+    });
+
+    expect(page.root).toMatchSnapshot();
+    expect(page.rootInstance.selectedTab).toBe(2);
+
+    const tabGroup = page.root.querySelector("ic-tab-group");
+
+    tabGroup.dispatchEvent(
+      new window.window.KeyboardEvent("keydown", {
+        key: "Home",
+        bubbles: true,
+        cancelable: true,
+      })
+    );
+
+    await page.waitForChanges();
+    expect(page.rootInstance.selectedTab).toBe(2);
+  });
+
+  it("should render tab context with tabs - manual activation", async () => {
+    const page = await newSpecPage({
+      components: [TabContext, TabGroup, Tab, TabPanel],
+      html: `<ic-tab-context activation-type="manual">
+      <ic-tab-group label="Example tab group">
+        <ic-tab>One</ic-tab>
+        <ic-tab>Two</ic-tab>
+        <ic-tab>Three</ic-tab>
+      </ic-tab-group>
+      <ic-tab-panel>Tab One</ic-tab-panel>
+      <ic-tab-panel>Tab Two</ic-tab-panel>
+      <ic-tab-panel>Tab Three</ic-tab-panel>
+      </ic-tab-context>`,
+    });
+
+    expect(page.root).toMatchSnapshot();
+
+    //test disconnected callback for manual activation
+    page.setContent("");
+  });
+
+  it("should render tab context with custom contextId", async () => {
```

Comment:
Is this test missing a 'toMatchSnapshot()'? Just that it doesn't seem to be testing the contextId at the moment